### PR TITLE
resource_control: add new privilege `RESOURCE_GROUP_USER`

### PR DESCRIPTION
**Pull request by glorv**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17681
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: optimizer-hints.md [hunk 1]
```diff
@@ -858,6 +858,10 @@ Example:
 SELECT /*+ RESOURCE_GROUP(rg1) */ * FROM t limit 10;
 ```
 
+> **Note:**
+>
+> From version v8.2.0, TiDB added privilege control for this hint. When system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode--new-in-v820) is set to `ON`, you need to have `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to use this hint. If you don't have these privileges, this hint will be ignored and a warning is returned, you can execute `SHOW WARNINGS;` to see the details.
```
Comment (lilin90) [inline optimizer-hints.md:863]:
```suggestion
> Starting from v8.2.0, TiDB introduces privilege control for this hint. When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode-new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to use this hint. If you do not have the required privilege, this hint is ignored, and TiDB returns a warning. You can view the details by executing `SHOW WARNINGS;` after the query.
```

File: privilege-management.md [hunk 2]
```diff
@@ -413,6 +413,10 @@ Requires `SUPER` or `RESOURCE_GROUP_ADMIN` privilege.
 
 Requires `SUPER` or `RESOURCE_GROUP_ADMIN` privilege.
 
+### SET RESOURCE GROUP
+
+When system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode--new-in-v820) is set to `ON`, you need to have `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to execute this statement.
```
Comment (lilin90) [inline privilege-management.md:418]:
```suggestion
When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode-new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to execute this statement.
```

File: sql-statements/sql-statement-set-resource-group.md [hunk 3]
```diff
@@ -24,6 +24,13 @@ ResourceGroupName ::=
 |   "DEFAULT"
 ```
 
+## Privilege
+
+This statement requires following configuration and privilege:
```
Comment (lilin90) [inline sql-statements/sql-statement-set-resource-group.md:29]:
```suggestion
Executing this statement requires the following configuration and privilege:
```

File: sql-statements/sql-statement-set-resource-group.md [hunk 4]
```diff
@@ -24,6 +24,13 @@ ResourceGroupName ::=
...
+
+1. System variable [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control--new-in-v660) is set to `ON`.
+2. When system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode--new-in-v820) is set to `ON`, privilege `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` is needed; when set to `OFF`, none of these privileges are needed.
```
Comment (lilin90) [inline sql-statements/sql-statement-set-resource-group.md:32]:
```suggestion
1. The system variable [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control-new-in-v660) is set to `ON`.
2. When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode-new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege; when it is set to `OFF`, none of these privileges are required.
```

File: system-variables.md [hunk 5]
```diff
@@ -5001,6 +5001,16 @@ SHOW WARNINGS;
 - Possible values: `"ddl"`, `"stats"`, `"br"`, `"lightning"`, `"background"`
 - This variable is used to explicitly specify the task type for the current session, which is identified and controlled by [Resource Control](/tidb-resource-control.md). For example: `SET @@tidb_request_source_type = "background"`.
 
+### tidb_resource_control_strict_mode <span class="version-mark">New in v8.2.0</span>
+
+- Scope: GLOBAL
+- Persists to cluster: Yes
+- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
+- Type: Boole
```
Comment (lilin90) [inline system-variables.md:5009]:
```suggestion
- Type: Boolean
```

File: system-variables.md [hunk 6]
```diff
@@ -5001,6 +5001,16 @@ SHOW WARNINGS;
...
+- Default value: `ON`
+- This variable controls whether privilege contorl is applied to statement [`SET RESOURCE GROUP`](/sql-statements/sql-statement-set-resource-group.md) and optimizer hint [`RESOURCE_GROUP()`](/optimizer-hints.md#resource_groupresource_group_name). When set to `ON`, you need to have `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to change the bind resource group of the current session or current statement via these two ways. When set to `OFF`, none of these privileges are needed, this behavior is the same as older versions without this variable.
```
Comment (lilin90) [inline system-variables.md:5011]:
```suggestion
- This variable controls whether privilege control is applied to the [`SET RESOURCE GROUP`](/sql-statements/sql-statement-set-resource-group.md) statement and the [`RESOURCE_GROUP()`](/optimizer-hints.md#resource_groupresource_group_name) optimizer hint. When this system variable is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to change the bound resource group of the current session or current statement via these two ways. When it is set to `OFF`, none of these privileges are required, and the behavior is the same as earlier TiDB versions without this variable.
```

File: system-variables.md [hunk 7]
```diff
@@ -5001,6 +5001,16 @@ SHOW WARNINGS;
...
+- If your cluster is upgraded from an earlier version, the default value of `tidb_resource_control_strict_mode` is set to `OFF`.
```
Comment (lilin90) [inline system-variables.md:5012]:
```suggestion
- When you upgrade your TiDB cluster from an earlier version to v8.2.0 or later, the default value of this variable is set to `OFF`, which means this feature is disabled by default.
```

File: tidb-resource-control.md [hunk 8]
```diff
@@ -238,7 +238,9 @@ For more details, see [`ALTER USER ... RESOURCE GROUP`](/sql-statements/sql-stat
 
 #### Bind the current session to a resource group
 
-By binding a session to a resource group, the resource usage of the corresponding session is limited by the specified usage (RU).
+You can use [`SET RESOURCE GROUP`](/sql-statements/sql-statement-set-resource-group.md) statement to change the bind resource group of current session. By binding a session to a resource group, the resource usage of the corresponding session is limited by the specified usage (RU).
```
Comment (lilin90) [inline tidb-resource-control.md:241]:
```suggestion
You can use the [`SET RESOURCE GROUP`](/sql-statements/sql-statement-set-resource-group.md) statement to change the bound resource group of the current session. By binding a session to a resource group, the resource usage of the corresponding session is limited by the specified usage (RU).
```

File: tidb-resource-control.md [hunk 9]
```diff
@@ -238,7 +238,9 @@ For more details, see [`ALTER USER ... RESOURCE GROUP`](/sql-statements/sql-stat
...
+
+When system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode--new-in-v820) is set to `ON`, you need to have `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to execute this statement.
```
Comment (lilin90) [inline tidb-resource-control.md:243]:
```suggestion
When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode-new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to execute this statement.
```

File: tidb-resource-control.md [hunk 10]
```diff
@@ -250,6 +252,8 @@ SET RESOURCE GROUP rg1;
 
 By adding the [`RESOURCE_GROUP(resource_group_name)`](/optimizer-hints.md#resource_groupresource_group_name) hint to a SQL statement, you can specify the resource group to which the statement is bound. This hint supports `SELECT`, `INSERT`, `UPDATE`, and `DELETE` statements.
 
+When system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode--new-in-v820) is set to `ON`, you need to have `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to use this hint.
```
Comment (lilin90) [inline tidb-resource-control.md:255]:
```suggestion
When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode--new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to use this hint.
```

File: sql-statements/sql-statement-set-resource-group.md [hunk 11]
```diff
@@ -24,6 +24,13 @@ ResourceGroupName ::=
 |   "DEFAULT"
 ```
 
+## Privilege
+
+Executing this statement requires the following configuration and privilege:
+
+1. The system variable [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control-new-in-v660) is set to `ON`.
+2. When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode-new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege; when it is set to `OFF`, none of these privileges are required.
```
Comment (hfxsd) [inline sql-statements/sql-statement-set-resource-group.md:32]:
```suggestion
- The system variable [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control-new-in-v660) is set to `ON`.
- When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode-new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege; when it is set to `OFF`, none of these privileges are required.
```

Comment (lilin90) [reply sql-statements/sql-statement-set-resource-group.md:32]: The numbered list is used on purpose, to emphasize that both the two must be satisfied.

**Review by lilin90 — APPROVED** [review]
/lgtm

**Review by lilin90 — COMMENTED** [review]
/approve

File: tidb-resource-control.md [hunk 12]
```diff
@@ -250,6 +252,8 @@ SET RESOURCE GROUP rg1;
 
 By adding the [`RESOURCE_GROUP(resource_group_name)`](/optimizer-hints.md#resource_groupresource_group_name) hint to a SQL statement, you can specify the resource group to which the statement is bound. This hint supports `SELECT`, `INSERT`, `UPDATE`, and `DELETE` statements.
 
+When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode--new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to use this hint.
```
Comment (lilin90) [inline tidb-resource-control.md:255]:
```suggestion
When the system variable [`tidb_resource_control_strict_mode`](/system-variables.md#tidb_resource_control_strict_mode-new-in-v820) is set to `ON`, you need to have the `SUPER` or `RESOURCE_GROUP_ADMIN` or `RESOURCE_GROUP_USER` privilege to use this hint.
```

**Pull request merged**